components/config-model-migrator/org.wso2.is.configuration.diff.creator/src/main/java/org/wso2/is/configuration/diff/creater/ConfigurationDiffChecker.java

```diff
@@ -0,0 +1,356 @@
+                    getDifferenceToMaps(migratedXMLFiles.get(entry.getKey()), detailedDiff, existingTags, keyValueMap
+                            , outputGenerator);
+                } else {
+                    filterUnTemplatedFiles(defaultXMLFiles, migratedXMLFiles, logFile, entry);
+                }
+            } catch (IOException e) {
+                log.error("Error occurred when parsing xml files or finding diff of xml files.");
+                throw new ConfigMigrateException("Error occurred when parsing xml files or finding diff of xml files" +
+                        ". ", e);
+            }
+        }
+    }
+
+    /**
+     * Check whether there is a file matched to the give file in .j2 file list. Because that tells that this file is
+     * templated or not.
+     *
+     * @param defaultXMLFiles Map of default config file list.
+     * @param j2files         Map of j2 file list.
+     * @param entry           Map entry of migrated config file list.
+     * @return true or false, if exists.
+     */
+    private boolean isFileTemplated(Map<String, File> defaultXMLFiles, Map<String, File> j2files, Map.Entry<String,
+            File> entry) {
+
+        String j2filename = entry.getKey() + MigrationConstants.J2_FILE_EXTENSION;
+        return defaultXMLFiles.get(entry.getKey()) != null && j2files.get(j2filename) != null;
+    }
+
+    /**
+     * Compare xml files using xmlUnit2 library.
+     *
+     * @param defaultFile Default xml file.
+     * @param migratedFile Migrated xml file.
+     * @return Diff of difference.
+     */
+    private Diff compareXMLFiles(File defaultFile, File migratedFile)  {
+
+        try {
+
+            return DiffBuilder.compare(migratedFile).withTest(defaultFile)
+                    .ignoreWhitespace()
+                    .ignoreComments()
+                    .checkForSimilar()
+                    .withNodeMatcher(new DefaultNodeMatcher(ElementSelectors.byName))
+                    .build();
+        } catch (XMLUnitException e) {
+            log.error("Error occurred when comparing default File : " + defaultFile.getName() + " and migrated" +
+                    " file :" + migratedFile.getName());
+        }
+        return null;
+    }
+
+    /**
+     * Compare difference to print the output.
+     *
+     * @param migratedFile    Migrated xml file.
+     * @param detailedDiff    Diff of all the difference.
+     * @param existingXMLTags Map of existing xml tags.
+     * @param keyValues       Map of key values.
+     * @param outputGenerator OutputGenerator.
+     */
+    private void getDifferenceToMaps(File migratedFile,
+                                     Diff detailedDiff,
+                                     Map<String, String> existingXMLTags,
+                                     Map<String, String> keyValues,
+                                     OutputGenerator outputGenerator) {
+
+        if (detailedDiff != null) {
```

Comment:
What happens if the detailedDiff is null?

Reply:
that means no diff, so no need to further analyse

Reply:
Then shall we have the null condition checked here? or we can have another private method to improve the readability.